mmdet/models/layers/transformer/utils.py

```diff
@@ -385,6 +439,297 @@ def inverse_sigmoid(x: Tensor, eps: float = 1e-5) -> Tensor:
     return torch.log(x1 / x2)
 
 
+class ConditionalAttention(BaseModule):
+    """A wrapper of conditional attention, dropout and residual connection.
+
+    Args:
+        embed_dims (int): The embedding dimension.
+        num_heads (int): Parallel attention heads.
+        attn_drop (float): A Dropout layer on attn_output_weights.
+            Default: 0.0.
+        proj_drop: A Dropout layer after `nn.MultiheadAttention`.
+            Default: 0.0.
+        cross_attn (bool): Whether the attention module is for cross attention.
+            Default: False
+        keep_query_pos (bool): Whether to transform query_pos before cross
+            attention.
+            Default: False.
+        batch_first (bool): When it is True, Key, Query and Value are shape of
+            (batch, n, embed_dim), otherwise (n, batch, embed_dim).
+             Default: True.
+        init_cfg (obj:`mmcv.ConfigDict`): The Config for initialization.
+            Default: None.
+    """
+
+    def __init__(self,
+                 embed_dims: int,
+                 num_heads: int,
+                 attn_drop: float = 0.,
+                 proj_drop: float = 0.,
+                 cross_attn: bool = False,
+                 keep_query_pos: bool = False,
+                 batch_first: bool = True,
+                 init_cfg=None):
+        super().__init__(init_cfg=init_cfg)
+
+        assert batch_first is True, 'First \
+        dimension of all DETRs in mmdet is \
+        `batch`, please set `batch_first` flag.'
```

Comment:
we do not support `batch_first=False` for ConditionalAttention
@LYMDLUT  @KeiChiTse  please help to fix this